docs/diagrams/wallet-lifecycle/wallet-lifecycle-1-handling-frauds.tikz

```diff
@@ -0,0 +1,187 @@
+\draw[->] (df_check2)  -| (df_check3)  \no;
+\draw[->] (df_check2)  -| (df_revert1) \yes;
+\draw[->] (df_check3)  -| (df_revert2) \no;
+\draw[->] (df_check3)  -| (df_action1) \yes;
+\draw[->] (df_action1) -- (df_action2);
+
+
+% DEFEAT FRAUD CHALLENGE WITH HEARTBEAT
+% Nodes
+\node[start, right=of df_start, xshift=110mm]                      (dfwt_start)   {\textbf{Defeat fraud challenge with heartbeat}};
+\node[check, below=of dfwt_start]                                  (dfwt_check1)  {Is there a pending\\ notification to be defeated?};
+\node[check, below right=of dfwt_check1]                           (dfwt_check2)  {Has the defeat timeout\\ been reported?};
+\node[check, below left=of dfwt_check2]                            (dfwt_check3)  {Is the\\ attatched message\\ a valid heartbeat\\ message?};
+\node[revert, below right=of dfwt_check2]                          (dfwt_revert1) {Transaction reverted};
+\node[revert, below left=of dfwt_check3]                           (dfwt_revert2) {Transaction reverted};
+\node[note1, right=of dfwt_start.south east, anchor=south west]    (dfwt_note1)   {Can be successfully called for wallets:\\ 
+                                                                                   \textbf{- LIVE\\
+                                                                                   - MOVING\_FUNDS\\
+                                                                                   - CLOSING}};
+\node[note2, above=of dfwt_note1.north, anchor=south, yshift=-8mm] (dfwt_note2)   {Sender incentivized by:\\ \textbf{Receiving recompensation from the treasury}};
+
+% Arrows
+% In places where arrows needs to bend more than once, we force the lines to
+% pass through intermediary points. 
+% (a,b) - sets point at an exact (a,b) coordinate
+% +(a,b) - sets point at current point and moves it by (a,b) vector, but doesn't
+%          remember the position afterwards
+% ++(a,b) - sets point at current point and moves it by (a,b) vector and
+%           remembers the position afterwards
+% To ease edition of the diagram in the future, usage of relative values is
+% recommended.
+\draw[->] (dfwt_start)   --                (dfwt_check1);
+\draw[->] (dfwt_check1)  -|                (dfwt_revert2) \no;
+\draw[->] (dfwt_check1)  -|                (dfwt_check2)  \yes;
+\draw[->] (dfwt_check2)  -|                (dfwt_check3)  \no;
+\draw[->] (dfwt_check2)  -|                (dfwt_revert1) \yes;
+\draw[->] (dfwt_check3)  -|                (dfwt_revert2) \no;
+\draw[->] (dfwt_check3)  -- +(4,0) \yes |- (df_action1);
+\draw[->] (df_action1)   --                (df_action2);
+
+
+% FRAUD DEFEAT TIMEOUT
+% Nodes
+\node[start, right=of dfwt_start, xshift=105mm]                   (fdt_start)   {Send notification about \textbf{fraud defeat timeout}};
+\node[check, below=of fdt_start]                                  (fdt_check1)  {Is there a pending\\ fraud notification that\\ hasn't been defeated?};
```

Comment:
Maybe:

> Is the given fraud challenge undefeated?

to clearly indicate this function is called for a single, particular fraud challenge.